### PR TITLE
completed lab 5

### LINKED ISSUE #285: Lab05 Stringer
- [ ] Create an executable go program in directory `05_stringer/luliangLi`
- [ ] Make the `IPAddr` type implement `fmt.Stringer` to print the address as a dotted quad
Find hints at [tour of go exercise: stringers](https://tour.golang.org/methods/18)
- [ ] Call `fmt.Println(IPAddr{127, 0, 0, 1})` in `main` to print:

### PR DESCRIPTION
**Pull request by luliangLi**
Fixes #285 

Review of colleague's PR #283 

#### Changes proposed in this PR:
- Complete lab 5 Stringers
- Test Stringers

### CODE REVIEW

**Review by n0npax — APPROVED** [review]
LGTM

**Comment by joshcarp** [comment]
@luliangLi Any reason why this looks almost identical to @n0npax lab5?

File: 05_stringer/luliangLi/main_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestStringify(t *testing.T) {
```
Comment (juliaogris) [inline 05_stringer/luliangLi/main_test.go:9]: See @n0npax's  [`TestStringify`](https://github.com/anz-bank/go-course/pull/215/files#diff-333f80df5d7b211e6addaa175bd579d7R25)

File: 05_stringer/luliangLi/main_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("127.0.0.1\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
```
Comment (juliaogris) [inline 05_stringer/luliangLi/main_test.go:42]: Identically error message to @n0npax's [`t.Errorf`](https://github.com/anz-bank/go-course/pull/215/files#diff-333f80df5d7b211e6addaa175bd579d7R21)

**Review by juliaogris — CHANGES_REQUESTED** [review]
We strongly encourage that you complete these labs on your own and don't copy them from others.   
Copying labs creates unnecessary effort on our side and limited learning on your side.

We don't know whether your solution coincidentally looks like #215 by @n0npax  (apologies!!), whether you have collaborated (acceptable, please clarify in PR description) or whether you have copied these lines (discouraged, please try again on your own). Just let us know what happened.

Thanks for your understanding.

**Comment by luliangLi** [comment]
> @luliangLi Any reason why this looks almost identical to @n0npax lab5?
@juliaogris @Joshcarp Hi, sorry for confusing you guys. I actually got the idea from the following stackoverflow: https://stackoverflow.com/questions/50568424/convert-a-byte-array-to-a-string-array, which is almost the same with the solution from @n0npax, which is coincidental I think.
For test cases file I actually copied from his, originally I would like to copy from my previous labs but it needed to switch git branches...

**Comment by juliaogris** [comment]
The go-course is now closed. Thank you very much for participating.

**Pull request closed without merging**